Auto-complete

Perdog opened the pull request.

GingkathFox commented:
Since i cant add to a pr (for some odd reason), heres some stuff to add:
`style.css`: set width to 300px in the `input[type="text"] {}` box, makes it easier to read the suggestions
`main.html`: turn delay to 10ms (makes it look faster xd), slow typers dont really need to worry since it takes about 1 second to get the item names, no need to delay it any more

Perdog commented:
- Increased the input width
- Added an array that caches search terms (Clears every time the program is closed)
- "suggestions" is now saved as the entire object, not just the name. This will let you use the item id for things later.
- Added the "renderItem" function because you now *have* to have it since the items are now objects, not just strings. As you can see it's just an HTML element that gets built in that function, so you can easily change the way it looks. (Like making the "highlight" on the actual search term yellow instead of the hard to see gray, for instance)
- Dropped the delay to 1 second. Trust me when I say you don't want to go lower then this. If you were using a local document or hard-written array in the script you could set it to 0, but because it's using API calls, if you have the delay set too low you run the risk of the app firing off 3-4 API calls as someone is typing. The main issue with that is the app doesn't care which call **happened** last, it cares about which call **returns** last. ----- Example: Someone is typing "tritan". If you set the delay too low, it could fire 4 times. Once for "tri", "trit", "trita", and "tritan". The last 3 will return fairly fast, there aren't many items with those search keys, however, "tri" is pretty broad, and will have a lot of matches, so it'll take a fraction longer then the others. **If*** it takes even a millisecond longer then the other searches to return, it's results will be displayed since that's the last one that returned. So your user will be looking at a big ass list of things that don't match the term in the search bar.

Perdog commented on Electron/main.html:
```diff
@@ -135,6 +135,7 @@
       </select><br>
       Item:<br>
       <input type="text" name="item" id="Item" autofocus placeholder="Tritanium" value=""><br><br>
+	  <!-- Aren't these backwards? -->
```
While playing around with your app I noticed a few 0.01 isk orders showing up when looking for sell orders. I think you might have your values backwards here.

GingkathFox replied:
If theyre sell orders, i think the 0.01 ISK orders are special corp prices (thats what i was told when i found those the first time)

Perdog commented on Electron/main.html:
```diff
@@ -135,6 +135,7 @@
       Item:<br>
       <input type="text" name="item" id="Item" autofocus placeholder="Tritanium" value=""><br><br>
+	  <!-- Aren't these backwards? -->
       <input type="radio" name="bOs" id="buyOrSell" value="buy">Who's Selling?<br>
```
```suggestion
      <input type="radio" name="bOs" id="buyOrSell" value="sell">Who's Selling?<br>
```

Perdog commented on Electron/main.html:
```diff
@@ -135,6 +135,7 @@
       <input type="text" name="item" id="Item" autofocus placeholder="Tritanium" value=""><br><br>
+	  <!-- Aren't these backwards? -->
       <input type="radio" name="bOs" id="buyOrSell" value="buy">Who's Selling?<br>
       <input type="radio" name="bOs" id="buyOrSell" value="sell">Who's Buying?<br><br>
```
```suggestion
      <input type="radio" name="bOs" id="buyOrSell" value="buy">Who's Buying?<br><br>
```

Perdog commented on Electron/Functions/getStations.js:
```diff
@@ -19,6 +20,9 @@ async function getStations(location) {
                     err(error, 'Function: getStations()')
                   })
+	} else {
+		promiseName = "Upwell market";
```
Unfortunately, there is no endpoint for Upwell structures that you can use unless you start asking for login tokens and all that crap, and I'm assuming you want to avoid that since you haven't done it up to this point. So this is where you can change the way they show up (I see you previously had it set as "private station", *technically* the upwells that show up when searching market orders aren't private, they're actually public stations that anyone can dock at, at the time the data was fetched.)

GingkathFox commented:
finally, ended up having to grab the pr branch, merge it using github desktop, and resolve conflict there
